pkg/healthcheck/healthcheck_multicluster.go

```diff
@@ -491,3 +539,178 @@ func (hc *HealthChecker) checkRemoteClusterGatewaysHealth(ctx context.Context) e
 	}
 	return &SkipError{Reason: "not checking muticluster"}
 }
+
+func (hc *HealthChecker) checkIfMirroredServicesHaveEndpoints() error {
+	if hc.Options.ShouldCheckMulticluster {
+
+		var servicesWithNoEndpoints []string
+		selector := fmt.Sprintf("%s, !%s", k8s.MirroredResourceLabel, k8s.MirroredGatewayLabel)
+		mirroredServices, err := hc.kubeAPI.CoreV1().Services(metav1.NamespaceAll).List(metav1.ListOptions{LabelSelector: selector})
+		if err != nil {
+			return err
+		}
+
+		for _, svc := range mirroredServices.Items {
+			// Check if there is a relevant end-point
+			_, err := hc.kubeAPI.CoreV1().Endpoints(svc.Namespace).Get(svc.Name, metav1.GetOptions{})
```

Comment:
Should we also check that the endpoints object is not empty?  i.e. that it contains an address?

Reply:
Hmm, Added an extra check here, as `len(endpoints.Subsets) == 0` but should the warning also be explicit that the endpoints is present but the `Subsets` is empty? 

I currently included that in the no endpoints found case.

Reply:
I think that's okay